src/control/server/server.go

```diff
@@ -172,6 +213,43 @@ func Start(log *logging.LeveledLogger, cfg *Configuration) error {
 		}
 	}
 
+	// Decide whether to use VMD devices.
+	if cfg.DisableVMD {
+		log.Debugf("VMD disabled in config (disable_vmd: true)")
+	} else {
+		if cfg.DisableVFIO {
+			return errors.New("VMD could not be enabled: VFIO disabled in config (disable_vfio: true)")
+		}
+		if !iommuDetected() {
+			return errors.New("VMD could not be enabled: IOMMU/VFIO not enabled")
+		}
+
+		vmdDevs, err := detectVmd()
+		if err != nil {
+			return errors.Wrap(err, "VMD could not be enabled")
+		}
+
+		// If VMD devices are going to be used, then need to run a separate
+		// bdev prepare (SPDK setup) with the VMD address as the PCI_WHITELIST
+		prepReqVmd := bdev.PrepareRequest{
+			PCIWhitelist:  strings.Join(vmdDevs, " "),
+			SkipReset:     true,
+			HugePageCount: prepReq.HugePageCount,
+			TargetUser:    prepReq.TargetUser,
+		}
```

Comment:
I think Sydney originally started with that approach but we will have to add fields to each request type (and vmd checks in each caller) rather than just setting the flag on the provider once. I can implement , but I'm not sure it will be preferable.

Reply:
From what I recall, the original approach required the admin to populate the server config with VMD addresses, and I agree that we don't want to expose the feature that way. My point is simply that it seems undesirable to add storage-specific functionality to the server package, as evidenced by the fact that this PR as it is currently requires a lot of invasive changes to support conveying VMD-specific info from the server to the bdev provider. The whole point of adding those sub-packages was to compartmentalize logic and avoid the need to make wide-ranging changes when adding new features.

Reply:
So maybe the correct addition to `bdev.PrepareRequest` isn't `VMDList`, but instead is a boolean set by the server config, e.g. `DisableVMD`. Then all of the code for managing VMD is moved into the bdev provider and is keyed off of the value of that bool.

Reply:
please review the most recent commits to see if it is acceptable